src/bin/mdev.rs

```diff
@@ -320,6 +374,23 @@ impl Opt {
     }
 }
 
+fn is_dir(path: &PathBuf) -> bool {
+    path.as_os_str().as_bytes().ends_with(&[b'/'])
+}
+
+fn replace_in_pathbuf(pb: &PathBuf, matches: &HashMap<OsString, OsString>) -> PathBuf {
```

Comment:
What would happen with an expansion `foo/%1%2` ?

Reply:
This code covers only single occurrences not interleaved to anything else, to cover cases like that or different cases like foo-%1, I this we'll need regexes to not fall for cases like reading %1 instead of %11

Reply:
let's look at the reference implementation and maybe document it first.

Reply:
I've been able to do the replacements without using regexes